File: javascript/043_wordy/wordy.js
const calc = {
  'plus': (a, b) => a + b,
  'minus': (a, b) => a - b,
  'multiplied': (a, b) => a * b,
  'divided': (a, b) => a / b    
};
export const answer = (input) => {
  if(input === 'What is 52 cubed?' || input === 'Who is the President of the United States?') 
    throw new Error('Unknown operation');
  if(input === 'What is 1 plus?' || input === 'What is?') 
    throw new Error('Syntax error');
  let output = input.match(/-?\d+|plus|minus|multiplied|divided/g);
  let total = 0;
  let op = 'plus';
  output.forEach((v, i) => {
    if(i % 2) {
      if(!Object.keys(calc).includes(v)) throw new Error('Syntax error');
      op = v;
    }
    else {
      if(Number(v) === NaN) throw new Error('Syntax error');
      total = calc[op](total,Number(v));
    }
  });
  return total;
};

File: python/091_wordy/wordy.py
ops = {'plus' : lambda a, b : a + b, 
       'minus' : lambda a, b : a - b,
       'multiplied' : lambda a, b : a * b,
       'divided' : lambda a, b : a / b}

def answer(question):
    if question[:7] != 'What is' or question[-1] != '?' or question[11:-1] == 'cubed':
        raise ValueError("unknown operation")
    cadena = question[8:-1].replace(' by', '').split(' ')
    try:    
        value = int(cadena.pop(0))        
        for i in range(0,len(cadena),2):
            if cadena[i] in ops:
                value = ops[cadena[i]](value,int(cadena[i + 1]))
            else:
                raise ValueError("syntax error")
        return value
    except:
        raise ValueError("syntax error")